docs/src/pages/errors.mdx

```diff
@@ -7,15 +7,51 @@
 You can customize the server-side behavior in your API handler's options by
 using an error formatter.
 
-By default, only the error message is returned to the client, to avoid leaking
-any sensitive information. You can customize this behavior by specifying the
-`errorFormatter` option when you initialize your file route helper. An error
-formatter runs on the server and takes the original `UploadThingError`, and
-returns a JSON-serializable object. The error also includes a `cause` property
-which contains more information about the nature of the error and what caused
-the error to throw in the first place.
-
-For example, if you're using Zod as an input parser, you can return information
+By default, the client only receives a generic message like `"Failed to run middleware"`
+to avoid leaking any sensitive information. You can customize this behavior
+by specifying the `errorFormatter` option when you initialize your file
+route helper. An error formatter runs on the server and takes the original 
+`UploadThingError`, and returns a JSON-serializable object. The error also
+includes a `cause` property which contains more information about the nature
+of the error and what caused the error to throw in the first place.
+
+You can also throw an `UploadThingError` inside your middleware
+which will send the error message to the client. All other error types will use
+a generic error message. Regardless of what error is thrown, you can still change
+the defaults with `errorFormatter`.
+
+```ts filename="server/uploadthing.ts"
+import { createUploadthing, type FileRouter } from "uploadthing/next";
+import { UploadThingError } from "uploadthing/server"
+
+const f = createUploadthing();
+
+const auth = (req: Request) => ({ id: "fakeId" });
+
+export const ourFileRouter = {
+  imageUploader: f({ image: { maxFileSize: "4MB" } })
+    .middleware(async ({ req }) => {
+      const user = await auth(req);
+
+      if (!user) throw new Error(`Cant find user from req: ${req.toString()}`); // client onError will get "Failed to run middleware"
+      if(!user.id) throw new UploadThingError('No user ID') // client onError will get "No user ID"
```

Comment:
maybe mention object constructor syntax too as that can be used to alter the http status code

Reply:
Yea so you could do like a 403 for example. Might wanna add a couple more generic ones though to make it truly usable